api/src/services/items.ts

```diff
@@ -217,7 +217,22 @@ export class ItemsService<Item extends AnyItem = AnyItem> implements AbstractSer
 					primaryKey = primaryKey ?? returnedKey;
 				}
 			} catch (err: any) {
-				throw await translateDatabaseError(err);
+				const dbError = await translateDatabaseError(err);
+
+				if (isDirectusError(dbError, ErrorCode.RecordNotUnique) && dbError.extensions.primaryKey) {
+					// This is a MySQL specific thing we need to handle here, since MySQL does not return the field name
+					// if the unique constraint is the primary key
+					dbError.extensions.field = pkField?.field ?? null;
+
+					if (dbError.extensions.collection === null) {
+						// This is a special case for MySQL 5.7, where the collection is not returned for the primary key constraint
```

Comment:
We do not support MySQL 5.7 any more, I think this can be dropped.

Reply:
You're correct. MySQL 5 is end of life completely https://endoflife.date/mysql